src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -102,16 +102,38 @@ public async Task SpecializeAsync()
         {
             _logger.LogInformation("Starting language worker channel specialization");
             _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+
             ILanguageWorkerChannel languageWorkerChannel = await GetChannelAsync(_workerRuntime);
-            if (_workerRuntime != null && languageWorkerChannel != null)
+            if (languageWorkerChannel != null)
             {
-                _logger.LogInformation("Loading environment variables for runtime: {runtime}", _workerRuntime);
-                await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
+                if (UsePlaceholderChannel(_workerRuntime))
+                {
+                    _logger.LogDebug("Loading environment variables for runtime: {runtime}", _workerRuntime);
+                    await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
+                }
+                else
+                {
+                    _logger.LogDebug("Shutting down placeholder worker. Worker is not compatible for runtime: {runtime}", _workerRuntime);
+                    // If we need to allow file edits, we should shutdown the webhost channel on specialization.
+                    await ShutdownChannelIfExistsAsync(_workerRuntime, languageWorkerChannel.Id);
+                }
             }
+
             _shutdownStandbyWorkerChannels();
             _logger.LogDebug("Completed language worker channel specialization");
         }
 
+        internal bool UsePlaceholderChannel(string workerRuntime)
+        {
+            if (!string.IsNullOrEmpty(workerRuntime))
```

Comment:
use as it is easier to read, if node check for run from zip
```
 internal bool UsePlaceholderChannel(string workerRuntime)
        {
            if (!string.IsNullOrEmpty(_workerRuntime) && string.Equals(_workerRuntime, LanguageWorkerConstants.NodeLanguageWorkerName))
            {
                return _environment.FileSystemIsReadOnly();
            }
            return true;
        }
```

Reply:
i this would fail for the case when _workerRuntime is null or empty, but we would return "true" for UsePlaceholderChannel. how does this look for being more readable? 

```
            if (!string.IsNullOrEmpty(workerRuntime))
            {
                // Special case: node apps must be read-only to use the placeholder mode channel
                // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
                if (string.Equals(_workerRuntime, LanguageWorkerConstants.NodeLanguageWorkerName))
                {
                    return _environment.FileSystemIsReadOnly();
                }
                return true;
            }
            return false;
```

Reply:
Looks good! yea please make the change